core/store/src/db.rs

```diff
@@ -524,11 +528,14 @@ impl RocksDBOptions {
 
     /// Opens a read only database.
     pub fn read_only<P: AsRef<std::path::Path>>(self, path: P) -> Result<RocksDB, DBError> {
-        let options = self.rocksdb_options.unwrap_or_default();
-        let cf_names: Vec<_> = self.cf_names.unwrap_or_else(|| vec!["col0".to_string()]);
-        let db = DB::open_cf_for_read_only(&options, path, cf_names.iter(), false)?;
-        let cfs =
-            cf_names.iter().map(|n| db.cf_handle(n).unwrap() as *const ColumnFamily).collect();
+        use strum::IntoEnumIterator;
+        let options = self.rocksdb_options.unwrap_or_else(rocksdb_options);
+        let cf_with_opts = DBCol::iter().map(|col| (col_name(col), rocksdb_column_options(col)));
```

Comment:
Feels a bit strange that we do `elf.cf_names.unwrap_or_else(|| DBCol::iter().map(|col| col_name(col)).collect());` for rw but just `DBCol::iter().map(|col| (col_name(col), rocksdb_column_options(col)));` for ro

Reply:
Sorry, I was unclear: I meant that it feels strange that we ignore `self.cf_names` and just go for all the columns.